src/java.base/share/classes/jdk/internal/foreign/BufferStack.java

```diff
@@ -165,10 +165,18 @@ private void assertOrder() {
             @ForceInline
             @Override
             @SuppressWarnings("restricted")
-            public MemorySegment allocate(long byteSize, long byteAlignment) {
+            public NativeMemorySegmentImpl allocate(long byteSize, long byteAlignment) {
+                // Do not zero initialize, so override the default zeroing method
+                // Todo: Maybe we could introduce a parameter that indicates zeroing
+                return allocateNoInit(byteSize, byteAlignment);
+            }
+
+            @ForceInline
+            @Override
+            public NativeMemorySegmentImpl allocateNoInit(long byteSize, long byteAlignment) {
```

Comment:
This probably also needs `@SuppressWarnings("restricted")`:
```suggestion
            @ForceInline
            @Override
            @SuppressWarnings("restricted")
            public NativeMemorySegmentImpl allocateNoInit(long byteSize, long byteAlignment) {
```

Reply:
I can't see why that is needed? Which restricted method is called by this method? What did I miss?